src/exceptions/clientException.ts

```diff
@@ -1,29 +1,24 @@
+import { AxiosError } from 'axios';
 import { Exception } from './exception';
 
 export class ClientException extends Exception {
   private static readonly statusCodeMap: Record<number, number> = {
     400: 422,
-    401: 401,
-    403: 403,
     404: 422,
   };
 
-  constructor(serviceName: string, error?: unknown) {
-    super('Dependent service returned error', ClientException.convertStatusCode(error as object), {
+  constructor(serviceName: string, error: AxiosError) {
+    super('Dependent service returned error', ClientException.convertStatusCode(error), {
```

Comment:
This still won't help btw... the `details` object is only in the superclass `Exception` where it can still be `any`. One way around would be adding a property here called e.g. `axiosError: AxiosError` ...

Then in consuming catch block, we can do:
```
if (error instanceOf ClientException && error.axiosError.response.status === 404)
...
```
Thoughts?

Reply:
I like that. There should also be other ways - like casting the details to AxiosError nad doing ` error.details?.response?.status === 404` but IDK what's the best. Also `AxiosError` has a `isAxiosError: bool` field if that helps in any way... Up to you!

Reply:
I don't want to cast it to AxiosError in every catch :/ But yeah, we can write type predicate using the `isAxiosError` and then always just do
```
if (error is AxiosErrior && error.details?.response?.status === 404)
...
```
In that way we don't need to even typecheck for `ClientException`. Maybe that is the "silver bullet" approach? ;P 
```

Reply:
Yeah that looks good! And there's still the way of having the extra `originalStatusCode` property.. I'm starting to like that as it's not bound to Axios

Reply:
See my other comment, that makes this more than `originalStatusCode` .. our custom HTTP error object.